keps/sig-network/3241-srv-service-name/README.md

```diff
@@ -0,0 +1,987 @@
+check that the given data are reflected in the resulting Endpoints
+and EndpointSlice objects.
+
+Negative tests to ensure that invalid values are rejected shall be
+implemented.
+
+The test suites for other modified components (e.g. kubernetes/dns)
+shall likewise be extended to test the new behaviour.
+
+### Graduation Criteria
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, [feature gate] graduations, or as
+something else. The KEP should keep this high-level with a focus on what
+signals will be looked at to determine graduation.
+
+Consider the following in developing the graduation criteria for this enhancement:
+- [Maturity levels (`alpha`, `beta`, `stable`)][maturity-levels]
+- [Feature gate][feature gate] lifecycle
+- [Deprecation policy][deprecation-policy]
+
+Clearly define what graduation means by either linking to the [API doc
+definition](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning)
+or by redefining what graduation means.
+
+In general we try to use the same stages (alpha, beta, GA), regardless of how the
+functionality is accessed.
+
+[feature gate]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+Below are some examples to consider, in addition to the aforementioned [maturity levels][maturity-levels].
+
+#### Alpha
+
+- Feature implemented behind a feature flag
+- Initial e2e tests completed and enabled
+
+#### Beta
+
+- Gather feedback from developers and surveys
+- Complete features A, B, C
+- Additional tests are in Testgrid and linked in KEP
+
+#### GA
+
+- N examples of real-world usage
+- N installs
+- More rigorous forms of testing—e.g., downgrade tests and scalability tests
+- Allowing time for feedback
+
+**Note:** Generally we also wait at least two releases between beta and
+GA/stable, because there's no opportunity for user feedback, or even bug reports,
+in back-to-back releases.
+
+**For non-optional features moving to GA, the graduation criteria must include
+[conformance tests].**
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+
+#### Deprecation
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality that deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+-->
+
+This adds a new optional attribute to the stable Service object API.
+There is no need for feature gating or a graduation process.  This
+field will be added to the existing API versions.
```

Comment:
you do need to go through all the process :smile: , also think about upgrades, downgrades and skew components:

- client X without the field against server X+1 with the field
- client X+1 with the field against server X without the field
- ...

Reply:
OK, I followed the approach from the `appProtocol` KEP (see https://github.com/kubernetes/enhancements/pull/1422/files#diff-0a58ac4551e1f218facd32aa731cb4151f846c049f4020a94cf61b7a12163b76R101-R105) - but I see now that the final KEP did use a feature gate.  I will update this proposal accordingly.

Reply:
As for the upgrade, downgrade and skew considerations, I have already addressed these.  If you feel it is lacking detail or I have overlooked something please be specific about the deficiency.